extensions/sql-database-projects/src/tools/shellExecutionHelper.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'promisify-child-process';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+export interface ShellCommandOptions {
+	workingDirectory?: string;
+	additionalEnvironmentVariables?: NodeJS.ProcessEnv;
+	commandTitle?: string;
+	argument?: string;
+}
+
+export class ShellExecutionHelper {
+	constructor(protected _outputChannel: vscode.OutputChannel) {
+	}
+
+	/**
+	 * spawns the dotnet command with arguments and redirects the error and output to ADS output channel
+	 */
+	public async runStreamedCommand(command: string, outputChannel: vscode.OutputChannel, options?: ShellCommandOptions): Promise<string> {
+		const stdoutData: string[] = [];
```

Comment:
I also added a similar function to utils. we should combine them. the code looks duplicate to me

Reply:
Ah, cool!  Lemme see what I want to do about that...

Reply:
I think I want to tackle this in another PR to make extra sure I don't break anything when combining.  The main difference looks like `child_process.spawn()` vs `child_process.exec()`, but there may be something else hiding in how the outputs are handled.

Reply:
Made https://github.com/microsoft/azuredatastudio/issues/17088